### PR TITLE
dffml: plugins: Added OS check for autosklearn

### LINKED ISSUE #887: dffml: plugins: Autosklearn should not be added for Windows OS
Just went through this: https://automl.github.io/auto-sklearn/master/installation.html#windows-osx-compatibility

It's an easy fix, to check for the os and then add it to the plugins list.

### PR DESCRIPTION
**Pull request by SGeetansh**
Added an OS check to exclude/include autosklearn into the core plugins. Fixes[ #887](https://github.com/intel/dffml/issues/887).

### CODE REVIEW

**Comment by yashlamba** [comment]
You can remove model/autosklearn from the below locations to test it with CI.
https://github.com/intel/dffml/blob/8087819432a187073d2acdde07e02ff8703bfc19/.github/workflows/testing.yml#L196
and
https://github.com/intel/dffml/blob/8087819432a187073d2acdde07e02ff8703bfc19/.github/workflows/testing.yml#L233

**Comment by SGeetansh** [comment]
@yashlamba Okay.

File: dffml/plugins.py [hunk 1]
```diff
@@ -45,6 +45,11 @@ def inpath(binary):
         ("model", "daal4py"),
     ]
 
+if platform.system() in {"Windows", "Darwin"}:
```
Comment (yashlamba) [inline dffml/plugins.py:48]:
```suggestion
if platform.system() not in {"Windows", "Darwin"}:
```

**Comment by yashlamba** [comment]
You'll need to add similar checks here again:
https://github.com/intel/dffml/blob/8087819432a187073d2acdde07e02ff8703bfc19/dffml/plugins.py#L102-L123

**Review by yashlamba — APPROVED** [review]
Looks good!

File: dffml/plugins.py [hunk 2]
```diff
@@ -110,7 +115,8 @@ def check_boost():
         "swig": lambda: inpath("swig"),
         "cython": lambda: inpath("cython"),
     }
-    if not python_package_installed("auto_sklearn")
+    if platform.system() not in {"Windows", "Darwin"}
+    and not python_package_installed("auto_sklearn")
```
Comment (pdxjohnny) [inline dffml/plugins.py:119]:
```suggestion
    and not python_package_installed("autosklearn")
```

Package name has no `_`

**Review by pdxjohnny — CHANGES_REQUESTED** [review]

**Comment by pdxjohnny** [comment]
Thank you! I must have forgot to hit the merge button the other day when I looked at this. Sorry for the delay!

**Pull request merged**